tests/test_utils.py

```diff
@@ -128,6 +131,103 @@ def test_segtowindow(self):
+        self.assertEqual(ap.utils.conversions.units.mag_to_flux(1.,0., mage=None), (10**(-1/2.5)), "mag incorrectly converted to flux (no error)")
+
+        #mag to flux with error:
+        self.assertEqual(ap.utils.conversions.units.mag_to_flux(1.,0., mage=1.), (10**(-1./2.5), np.log(10)*(1./2.5)*10**(-1./2.5)), "mag incorrectly converted to flux (with error)")
+        
+        #magperarcsec2 to mag with area A defined
+        self.assertEqual(ap.utils.conversions.units.magperarcsec2_to_mag(1., a=None, b=None, A=1.), (1. - 2.5 * np.log10(1.)), "mag/arcsec^2 incorrectly converted to mag (area A given, a and b not defined)")
+
+        #magperarcsec2 to mag with semi major and minor axes defined (a, and b)
+        self.assertEqual(ap.utils.conversions.units.magperarcsec2_to_mag(1., a=1., b=1., A=None), (1. - 2.5 * np.log10(np.pi)), "mag/arcsec^2 incorrectly converted to mag (semi major/minor axes defined)")
+
+        #mag to magperarcsec2 with area A defined
+        self.assertEqual(ap.utils.conversions.units.mag_to_magperarcsec2(1., a=None, b=None, A=1., R=None), (1. + 2.5 * np.log10(1.)), "mag incorrectly converted to mag/arcsec^2 (area A given)")
+
+        #mag to magperarcsec2 with radius R given (assumes circular)
+        self.assertEqual(ap.utils.conversions.units.mag_to_magperarcsec2(1., a=None, b=None, A=None, R=1.), (1. + 2.5 * np.log10(np.pi)), "mag incorrectly converted to mag/arcsec^2 (radius R given)")
+
+        #mag to magperarcsec2 with semi major and minor axes defined (a, and b)
+        self.assertEqual(ap.utils.conversions.units.mag_to_magperarcsec2(1., a=1., b=1., A=None, R=None), (1. + 2.5 * np.log10(np.pi)), "mag incorrectly converted to mag/arcsec^2 (area A given)")
+
+        #position angle PA to radians
+        self.assertEqual(ap.utils.conversions.units.PA_shift_convention(1., unit='rad'), ((1. - (np.pi / 2)) % np.pi), "PA incorrectly converted to radians")
+
+        #position angle PA to degrees
+        self.assertEqual(ap.utils.conversions.units.PA_shift_convention(1., unit='deg'), ((1. - (180 / 2)) % 180), "PA incorrectly converted to degrees")
+
+
+    def test_conversion_dict_to_hdf5(self):
+
+        #convert string to hdf5
+        self.assertEqual(ap.utils.conversions.dict_to_hdf5.to_hdf5_has_None(l='test'), (False), "Failed to properly identify string object while converting to hdf5")
+
+        #convert __iter__ to hdf5
+        self.assertEqual(ap.utils.conversions.dict_to_hdf5.to_hdf5_has_None(l='__iter__'), (False), "Attempted to convert '__iter__' to hdf5 key")
+
+        #convert hdf5 file to dict
+        h = h5py.File("mytestfile.hdf5", "w")
+        dset = h.create_dataset("mydataset", (1,), dtype='i')
+        dset[...] = np.array([1.0])
+        self.assertEqual(ap.utils.conversions.dict_to_hdf5.hdf5_to_dict(h=h), ({'mydataset': h['mydataset']}), "Failed to convert hdf5 file to dict")
+
+        #convert dict to hdf5
+        d = {'mydata1': 'statement', 'mydata2': 'statement2'}
```

Comment:
Perhaps instead of a generic dictionary here, you could make a simple model (e.g. sersic with manual parameter values) and save it to an hdf5 file? Should be able to just call `model.save("mytestfile.hdf5")` then you could test that file to ensure that the correct values are recovered.

Reply:
Hey Connor, good idea. I was just trying for something fast! But I'll implement that.